src/modules/commander/commander.cpp

```diff
@@ -1737,6 +1740,19 @@ Commander::run()
 		_local_position_sub.update();
 		_global_position_sub.update();
 
+		// Set the allowable positon uncertainty based on combination of flight and estimator state
+		// When we are in a operator demanded position control mode and are solely reliant on optical flow, do not check position error becasue it will gradually increase throughout flight and the operator will compensate for the drift
+		bool reliant_on_opt_flow = ((estimator_status.control_mode_flags & (1 << estimator_status_s::CS_OPT_FLOW))
+					    && !(estimator_status.control_mode_flags & (1 << estimator_status_s::CS_GPS))
+					    && !(estimator_status.control_mode_flags & (1 << estimator_status_s::CS_EV_POS)));
+		bool operator_controlled_position = (internal_state.main_state == commander_state_s::MAIN_STATE_POSCTL);
+		_skip_pos_accuracy_check = reliant_on_opt_flow && operator_controlled_position;
```

Comment:
If the threshold is now infinity in this position control + optical flow state isn't it safe to continue running the checks? That way we'd still catch a global position timeout or stale data, and it simplifies the logic.